### PR TITLE
Gives virus produced reagents their own area to avoid future chemical conflicts/mishaps

### PR DESCRIPTION
**Pull request by LanCartwright**
WIP, will update further.

### CODE REVIEW

File: code/modules/reagents/chemistry/reagents/virology_reagents.dm [hunk 1]
```diff
@@ -0,0 +1,241 @@
+
+//////////////////////////////////////////////////////////////////////////////////////////
+					// VIROLOGY REAGENTS
+//////////////////////////////////////////////////////////////////////////////////////
+
+// where all the reagents related to medicine go.
```
Comment (Supermichael777) [inline code/modules/reagents/chemistry/reagents/virology_reagents.dm]: copypasta detected

Comment (LanCartwright) [inline code/modules/reagents/chemistry/reagents/virology_reagents.dm]: Huh.

Anyway, //most// of these are basically copy pasta versions of themselves.

**Comment by Iamgoofball** [comment]
:+1: allows us to change reagent effects without viruses also changing

File: code/modules/reagents/chemistry/reagents/virology_reagents.dm [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+	color = "#C8A5DC"
+
+/datum/reagent/medicine/mannitol/on_mob_life(mob/living/M)
+	M.adjustBrainLoss(-3*REM)
+	..()
+
+/datum/reagent/virology/morphine
+	name = "Opiates"
+	id = "viromorphine"
+	description = "A nerf blocker that causes drowsiness and eventually unconsciousness in high doses. Overdose will cause a variety of effects, ranging from minor to lethal."
```
Comment (Fox-McCloud) [inline code/modules/reagents/chemistry/reagents/virology_reagents.dm]: `nerf` blocker.

Comment (LanCartwright) [inline code/modules/reagents/chemistry/reagents/virology_reagents.dm]: nerve*

**Comment by phil235** [comment]
that's a lot of copypasta :confused:

**Comment by LanCartwright** [comment]
eh, I'm sorry phil, but they're really meant to be copies of their last stable chemical, with some minor adjustments. I didn't really want to make new ones for each when they already had their functions.

**Comment by optimumtact** [comment]
don't viruses already have an on mob life? they should be able to do this stuff

**Comment by LanCartwright** [comment]
This was done to ensure nothing can ever affect virology chems unless chems here are changed. This is for safety reasons. I'll work to change any remaining ones that I couldn't cover here once this is merged to avoid conflicts.

**Comment by Fox-McCloud** [comment]
>don't viruses already have an on mob life? they should be able to do this stuff rather than relying on reagents to do it.

VIruses aren't processed in the mob loop for some reason; they have their entire own controller for some reason. Don't know why.

**Comment by optimumtact** [comment]
They probably should so people stop shoehorning reagents into viruses to make them have effects on people.

**Comment by LatD** [comment]
Does this also fix #17457 ?

**Comment by KorPhaeron** [comment]
I spoke with J_mad, he's gonna try making the viruses not use chems

**Pull request closed without merging**